### PR TITLE
crdt collection tests

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: src/javaharness/java/arcs/crdt/CollectionDataTest.java [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    ));
+    assert !set.applyOperation(new CollectionOperation<Data>(
+      CollectionOpTypes.REMOVE,
+      new Data("one"),
+      new VersionMap() {{ put("me", 0); }},
+      "me"
+    ));
+  }
+
+  public static void testRejectRemoveNonexitent() {
```
Comment (shaper) [inline src/javaharness/java/arcs/crdt/CollectionDataTest.java]: s/exitent/existent/ here and below where referenced

File: src/runtime/crdt/tests/crdt-collection-test.ts [hunk 2]
```diff
@@ -222,3 +222,5 @@ describe('CRDTCollection', () => {
     assert.deepEqual(modelChange2, otherChange2);
   });
 });
+
+// TODO: if/when adding more tests to this file, please, also update CollectionDataTest.java
```
Comment (shaper) [inline src/runtime/crdt/tests/crdt-collection-test.ts]: Perhaps we should file a P2 issue to automate this -- we could add something to sigh that at least outputs a warning or something, when a file in a whitelist is modified in a PR, to remind the author to not forget to change the corresponding file. And vice versa with the Java side.

File: src/javaharness/java/arcs/crdt/CollectionDataTest.java (hunk 1 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/crdt/CollectionDataTest.java]: done

File: src/runtime/crdt/tests/crdt-collection-test.ts (hunk 2 above)
Comment (mmandlis) [reply src/runtime/crdt/tests/crdt-collection-test.ts]: i added an equivalent note to the java file. 
about the automation - this won't apply to only tests then, but also all the parallel implementation files, will it? that's quite a lot of files, and i also doubt that they'll be always changed in the same pr.. so at least until the java implementation is more mature, i'm not sure the automation will be really helpful.

**Pull request merged**